lib/pana.dart

```diff
@@ -40,8 +40,12 @@ class PackageAnalyzer {
         dartSdk: _dartSdk, flutterSdk: _flutterSdk, pubCacheDir: pubCacheDir);
   }
 
-  Future<Summary> inspectPackage(String package,
-      {String version, bool keepTransitiveLibs: false}) async {
```

Comment:
Wondering if this should be refactored – so there is a method that just operates off a package in a directory – then wrap it in method that does the download to cache?

Reply:
For that I'd slightly modify the PubSummary as it is now: the `pubspec.yaml` parsing should not be blocked on the pub output parsing and conflict detection (there are a few throws there if there is a conflict somewhere). But other than that, I think the two method should be good to go.

Reply:
Hm, it is a bit more than just having a new method there, I'd rather do it in a separate PR.